test/cfg/boost.cpp

```diff
@@ -104,4 +105,24 @@ void lock_guard_finiteLifetime(boost::mutex& m)
 {
     // cppcheck-suppress unusedScopedObject
     boost::lock_guard<boost::mutex>{ m };
-}
\ No newline at end of file
+}
+
+BOOST_AUTO_TEST_SUITE(my_auto_test_suite)
+
+BOOST_AUTO_TEST_CASE(test_message_macros)
+{
+    bool my_bool = false;
+    BOOST_WARN_MESSAGE(my_bool, "warn");
+    BOOST_CHECK_MESSAGE(my_bool, "check");
+    BOOST_REQUIRE_MESSAGE(my_bool, "require");
+
+    BOOST_WARN_MESSAGE(my_bool, "my_bool was: " << my_bool);
+}
+
+using test_types_w_tuples = std::tuple<int, long, unsigned char>;
+BOOST_AUTO_TEST_CASE_TEMPLATE(my_tuple_test, T, test_types_w_tuples)
+{
+    BOOST_TEST(sizeof(T) == 4U);
```

Comment:
CI complains about T here.

Reply:
Unfortunately unrolling the type list of the `BOOST_AUTO_TEST_CASE_TEMPLATE` macro would be quite complicated. I therefore decided to suppress the warning in this test